generated/nidigital/nidigital/_grpc_stub_interpreter.py

```diff
@@ -0,0 +1,655 @@
+# -*- coding: utf-8 -*-
+# This file was generated
+
+import grpc
+import hightime  # noqa: F401
+import threading
+import warnings
+
+from . import enums as enums
+from . import errors as errors
+from . import nidigitalpattern_pb2 as grpc_types
+from . import nidigitalpattern_pb2_grpc as nidigital_grpc
+from . import session_pb2 as session_grpc_types
+
+from . import history_ram_cycle_information as history_ram_cycle_information  # noqa: F401
+
+
+class GrpcStubInterpreter(object):
+    '''Interpreter for interacting with a gRPC Stub class'''
+
+    def __init__(self, grpc_options):
+        self._grpc_options = grpc_options
+        self._lock = threading.RLock()
+        self._client = nidigital_grpc.NiDigitalStub(grpc_options.grpc_channel)
+        self.set_session_handle()
+
+    def set_session_handle(self, value=session_grpc_types.Session()):
```

Comment:
Where is this class defined? I don't see it in session_pb2.

Reply:
The _pb2 files are weird. Most of the stuff is instantiated from the data in the magic DESCRIPTOR string. You can see the Session definition in the .proto file, [here](https://github.com/ni/nimi-python/blob/726fad928e831f4b355919ea746d9ac89ca04e9c/src/shared_protos/session.proto#L36).